src/piracy.jl

```diff
@@ -123,6 +149,10 @@ function is_pirate(meth::Method)
         signature = signature.body
     end
 
+    # the first parameter in the signature is the function type, and it
+    # follows slightly other rules if it happens to be a Union type
+    is_foreign_method(signature.parameters[1], method_pkg) || return false
+
     all(param -> is_foreign(param, method_pkg), signature.parameters)
```

Comment:
As the first entry of `signature.parameters` is handled differently, the `is_foreign` check should only be applied to `signature.parameters[2:end]`, right?